reboot_toolkit/toolkit_backend.py

```diff
@@ -862,42 +1031,103 @@ def save_figs_to_html(
-#
-#
-# if __name__ == "__main__":
-#     main()
+def export_data(
+    reboot_client: RebootClient,
+    org_player_id: str,
+    movement_type_enum: MovementType,
+    data_type_enum: DataType,
+    session_ids: list[str],
+    verbose: bool = True,
+) -> pd.DataFrame:
+    """
+    Export data using the Reboot API.
+
+    :param reboot_client: RebootClient object
+    :param org_player_id: the org player ID to download data for
+    :param movement_type_enum: the movement type enum to download data for
+    :param data_type_enum: the data type enum to download
+    :param session_ids: list of session IDs to download
+    :param verbose: whether to print progress
+    :return: dataframe of returned data
+    """
+    data_format = "parquet"
+    aggregate = True
+    return_column_info = False
+    return_data = True
+    as_pyarrow = True
+    use_threads = False
+
+    with ThreadPoolExecutor() as executor:
+        pyarrow_tables = list(
+            tqdm(
+                executor.map(
+                    reboot_client.data_exports.create,
+                    session_ids,
+                    repeat(MOVEMENT_TYPE_IDS_MAP[movement_type_enum.value]),
+                    repeat(org_player_id),
+                    repeat(data_type_enum.value),
+                    repeat(data_format),
+                    repeat(aggregate),
+                    repeat(return_column_info),
+                    repeat(return_data),
+                    repeat(as_pyarrow),
+                    repeat(use_threads),
+                ),
+                total=len(session_ids),
+                disable=not verbose,
+            )
+        )
+
+    return pa.concat_tables(pyarrow_tables).to_pandas()
+
+
+def add_offsets_from_metadata(
+    data_df: pd.DataFrame, metadata_df: pd.DataFrame, movement_type_enum: MovementType
+) -> pd.DataFrame:
+    """
+    Add the X, Y, and Z offsets from the metadata to the position columns of an existing dataframe.
+
+    :param data_df: the data for adding the offsets
+    :param metadata_df: the dataframe containing the offsets
+    :param movement_type_enum: the movement type enum
+    :return: the dataframe with the added offsets
+    """
+    accepted_movement_types = {"baseball-hitting"}
+
+    if movement_type_enum.value not in accepted_movement_types:
+        raise NotImplementedError(
+            "Adding offsets only implemented for movement types: {}, others coming soon!".format(
+                accepted_movement_types
+            )
+        )
+
+    assert set(data_df["org_movement_id"]) == set(
+        metadata_df["org_movement_id"]
+    ), "Mismatched org movement ids in data_df and metadata_df, but must be matched to add offsets"
+
+    base_cols = list(data_df)
+
+    offset_cols = [
+        "org_movement_id",
+        "X_offset",
+        "Y_offset",
+        "Z_offset",
+    ]
+
+    data_df = data_df.merge(metadata_df[offset_cols], on="org_movement_id", how="left")
+
+    cols_to_drop = []

```

Comment:
The values for `cols_to_drop` are listed out in `offset_cols` along with org_movement_id, but then are being created by appending to a list here. It feels somewhat redundant.

Reply:
yeah i guess i did this as a way to exclude the org movement ID from the columns to drop, but could do this a little differently so it's not as redundant